### PR TITLE
[v9] Backport #12891

### PR DESCRIPTION
**Pull request by xacrimon**
Backport of #12891. This was a bit of a messy backport with some surgery involved. Tests pass and it seems to work when tested with AD.

### CODE REVIEW

File: lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs [hunk 1]
```diff
@@ -22,6 +23,7 @@ use rdp::core::{mcs, tpkt};
 use rdp::model::error::*;
 use rdp::try_let;
 use std::collections::HashMap;
+use std::collections::VecDeque;
```
Comment (espadolini) [inline lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs:26]: merge imports?

**Review by espadolini — APPROVED** [review]

File: lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs (hunk 1 above)
Comment (xacrimon) [reply lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs:26]: This was split by rustfmt :/

**Comment by zmb3** [comment]
I tried to run this and couldn't make a connection to the desktop.

Logs show:

```
[2022-06-13T14:48:26Z WARN  rustls::check] Received a ServerHelloDone handshake message while expecting [CertificateRequest]
```

I confirmed that connecting to the same desktop with master (4742327cff077efe59b7405ccb2ee3f9aec3a740) works fine.

@xacrimon @LKozlowski could you take a look?

**Comment by zmb3** [comment]
Interesting - I went back to this branch and tried again and now it is working, and I do _not_ see this rustls message in the logs when it works.

Plus, based on the changes here, I wouldn't expect these to cause issues with rustls.

Is our rustls out of date by chance though? Maybe an update could help?

**Comment by xacrimon** [comment]
@zmb3 As expected this is a transient issue due to rustls. I went ahead and made a semver compatible update of all deps using `cargo update`. It did update rustls from 0.20.4 to 0.20.6 which fixes this. Should be good to go now.

**Comment by xacrimon** [comment]
Does the unicode clipboard work too?

**Comment by zmb3** [comment]
@xacrimon Yep, works a charm.

**Comment by zmb3** [comment]
@xacrimon sorry, one more thing - since this was already merged to master, our Rust deps are probably out of sync. Would you mind opening a PR to master as well with the `cargo update` results?

**Comment by xacrimon** [comment]
Yep, @zmb3 could you approve this though?

**Review by zmb3 — APPROVED** [review]

**Pull request merged**